extensions/amp-list/0.1/amp-list.js

```diff
@@ -550,8 +589,14 @@ export class AmpList extends AMP.BaseElement {
     if (!elementSrc) {
       return Promise.resolve();
     }
+
     let fetch;
-    if (this.ssrTemplateHelper_.isEnabled()) {
+    if (this.isAmpStateSrc_(elementSrc)) {
```

Comment:
Why is this block added before the `ssrTemplateHelper_.isEnabled()` condition? Does this mean that amp-state can be used to init amp-list for SSR enabled documents?

Reply:
Correct.

This definition of SSR has two pieces:
- Fetches are delegated to the viewer
- Template rendering is also done by the viewer

In the local data case, there is no fetch, just a peek into amp-state. So there is no reason to delegate to the viewer (we already do this for bind expressions with `renderLocalData`)

Reply:
> Template rendering is also delegated to the viewer

I think this is an issue for affected email clients. SSR templates is needed for spam checking dynamic content. We'll need to discuss if/how we can reconcile this, but for now let's just make this an error case.

Note that the validator also checks URI schemes for `src`. We can defer that change until we're closer to launch.

Reply:
> I think this is an issue for affected email clients.

Isn't the correct behavior here already covered in `doRenderPass`?
https://github.com/ampproject/amphtml/blob/21d49e31b73193544e212e2910df84005c99138d/extensions/amp-list/0.1/amp-list.js#L786-L819

Reply:
It'll still fail but (1) downstream of the actual issue (too late) and (2) with a confusing error. 

Then again, this would actually fail in validation first so the email should never be sent in the first place. But we should try to do the most reasonable thing.

Reply:
Resolving this now. I was confused about two things:
1. I assumed that the first viewer communication just did the fetch and that `doRenderPass` then did template rendering, but the first one actually has the viewer do both.
2. Whether bind json has the same issue (it does).

Closing now.